EIPS/eip-2935.md

```diff
@@ -40,10 +41,10 @@ def process_block_hash_history(block: Block, state: State):
     if block.timestamp >= FORK_TIMESTAMP:
         state.insert_slot(HISTORY_STORAGE_ADDRESS, (block.number-1) % HISTORY_SERVE_WINDOW , block.parent.hash)
 
-    # If this is the first fork block, add the parent's direct 255 ancestors as well
+    # If this is the fork block, add the parent's direct `BLOCKHASH_PERSIST_WINDOW - 1` ancestors as well
     if block.parent.timestamp < FORK_TIMESTAMP:
         ancestor = block.parent
-        for i in range(HISTORY_SERVE_WINDOW - 1):
+        for i in range(FORK_TRANSITION_PERSIST_LENGTH - 1):
```

Comment:
I think you intended to put `HISTORY_SERVE_WINDOW`? since FORK_TRANSITION_PERSIST_LENGTH` doesn't exist. But still not sure this shouldn't be `HISTORY_SERVE_WINDOW` to migrate all the history, just to know all slots of the ring buffer are valid which might simplify data accessed from outside BLOCKHASH

Reply:
ahh its `BLOCKHASH_PERSIST_WINDOW` cleaned up